tw-tasks-core/src/main/java/com/transferwise/tasks/dao/MySqlTaskDao.java

```diff
@@ -562,86 +589,12 @@ public DeleteFinishedOldTasksResult deleteOldTasks(TaskStatus taskStatus, Durati
 
         processedIdsCount += bucketSize;
       }
-
     }
 
     result.setDeletedTasksCount(deletedTasksCount);
     result.setDeletedUniqueKeysCount(deletedUniqueTaskKeysCount);
     result.setDeletedTaskDatasCount(deleteTaskDatasCount);
-    result.setFoundTasksCount(taskIds.size());
-    result.setDeletedBeforeTime(TimeUtils.toZonedDateTime(deletedBeforeTime));
-
-    return result;
-  }
-
-  protected DeleteFinishedOldTasksResult deleteOldTasksParanoid(TaskStatus taskStatus, Duration age, int batchSize) {
-    DeleteFinishedOldTasksResult result = new DeleteFinishedOldTasksResult();
-    Timestamp deletedBeforeTime = Timestamp.from(Instant.now(TwContextClockHolder.getClock()).minus(age));
-
-    List<Pair<Object, Long>> taskVersionIds = jdbcTemplate.query(deleteFinishedOldTasksSql,
-        args(taskStatus.name(), deletedBeforeTime, batchSize),
-        (rs, rowNum) -> ImmutablePair.of(rs.getObject(1), rs.getLong(2)));
-
-    if (!taskVersionIds.isEmpty()) {
-      UUID firstDeletedTaskId = sqlMapper.sqlTaskIdToUuid(taskVersionIds.get(0).getLeft());
-      result.setFirstDeletedTaskId(firstDeletedTaskId);
-      ZonedDateTime nextEventTime = getFirst(jdbcTemplate.query(deleteFinishedOldTasksSql1,
-          args(firstDeletedTaskId),
-          (rs, rowNum) -> TimeUtils.toZonedDateTime(rs.getTimestamp(1))));
-
-      result.setFirstDeletedTaskNextEventTime(nextEventTime);
-    }
-
-    int tasksCount = 0;
-    int[] tasksDeleteResult = jdbcTemplate.batchUpdate(deleteTaskSql, new BatchPreparedStatementSetter() {
-      @Override
-      public void setValues(PreparedStatement ps, int i) throws SQLException {
-        ps.setObject(1, taskVersionIds.get(i).getLeft());
-        ps.setLong(2, taskVersionIds.get(i).getRight());
-      }
-
-      @Override
-      public int getBatchSize() {
-        return taskVersionIds.size();
-      }
-    });
-
-    List<Object> deletedIds = new ArrayList<>();
-    for (int i = 0; i < tasksDeleteResult.length; i++) {
-      if (tasksDeleteResult[i] > 0) {
```

Comment:
This does not work anymore with latest mariadb driver. It does not give back individual affected rows count.